src/aspire/image/image.py

```diff
@@ -198,24 +202,40 @@ def transpose(self):
 
         :return: Image instance.
         """
-        return Image(np.transpose(self._data, (0, 2, 1)))
+        original_stack_shape = self.stack_shape
+
+        im = self.stack_reshape(-1)
+        imt = np.transpose(im._data, (0, -1, -2))
+
+        return Image(imt).stack_reshape(original_stack_shape)
 
-    def flip(self, axis=1):
+    def flip(self, axis=-2):
         """
         Flip image stack data along axis using numpy.flip().
 
         :param axis: Optionally specify axis as integer or tuple.
-            Defaults to axis=1.
+            Defaults to axis=-2.
 
         :return: Image instance.
         """
-        if axis == 0 or (isinstance(axis, Iterable) and 0 in axis):
-            raise ValueError("Cannot flip axis 0: stack axis.")
+        # Convert integer to tuple, so we can always loop.
+        if isinstance(axis, int):
+            axis = (axis,)
+
+        # Check we are not attempting to flip any stack axis.
+        for ax in axis:
+            ax = ax % self.ndim  # modulo [0, ndim)
+            if ax < self.stack_ndim:
+                raise ValueError(
+                    f"Cannot flip axis {ax}: stack axis. Did you mean {ax-3}?"
+                )
 
         return Image(np.flip(self._data, axis))
 
     def __repr__(self):
-        return f"{self.n_images} images of size {self.resolution}x{self.resolution}"
+        msg = f"{self.n_images} {self.dtype} images of arranged as a {self.stack_shape} stack"
```

Comment:
Awesome. Small typo: "images ~~of~~ arranged..."

Reply:
Fixed